### PR TITLE
Remove type assertions from `prepare-googletag`

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Small health PR to remove type assertions.

Remove the `gu.commercial.pageViews` session storage, as it’s been unused since #19441.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

Less code, stronger types.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [ ] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts [hunk 1]
```diff
@@ -59,18 +59,24 @@ const setDfpListeners = (): void => {
...
 	}
 };
 
 const setPageTargeting = (): void => {
 	const pubads = window.googletag.pubads();
 	// because commercialFeatures may export itself as {} in the event of an exception during construction
-	const targeting = getPageTargeting() as Record<string, string | string[]>;
-	Object.keys(targeting).forEach((key) => {
-		pubads.setTargeting(key, targeting[key]);
+	const targeting = getPageTargeting();
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts:76]: V.minor - can we inline `pubads` and `targeting`? I don't think we wouldn't lose much readability?

**Review by arelra — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts [hunk 2]
```diff
@@ -59,18 +59,21 @@ const setDfpListeners = (): void => {
...
+		);
+
+		// Increment only if positive number and not NaN, otherwise set to 0
+		typeof pageViews === 'number' && pageViews >= 0
+			? storage.session.set('gu.commercial.pageViews', pageViews + 1)
+			: storage.session.set('gu.commercial.pageViews', 0);
 	}
 };
 
 const setPageTargeting = (): void => {
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts:73]: V minor again - I think we can do an implicit return now :)

**Review by chrislomaxjones — CHANGES_REQUESTED** [review]

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts [hunk 3]
```diff
@@ -59,18 +59,24 @@ const setDfpListeners = (): void => {
...
-			(storage.session.get('gu.commercial.pageViews') as number) || 0;
-		storage.session.set('gu.commercial.pageViews', pageViews + 1);
+		const pageViews: unknown = storage.session.get(
+			'gu.commercial.pageViews',
+		);
+
+		// Increment only if positive number and not NaN, otherwise set to 0
+		typeof pageViews === 'number' && pageViews >= 0
+			? storage.session.set('gu.commercial.pageViews', pageViews + 1)
+			: storage.session.set('gu.commercial.pageViews', 0);
```
[the remark this answers is not in the document]
Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts:71]: I think it should either increment if it’s a valid number, or set it to 0 if it’s missing or invalid.

[the remark this answers is not in the document]
Comment (chrislomaxjones) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts:71]: The original logic reads to me "retrieve pageview count from localstorage or set to zero if you cant, then save that number plus one", so was there a case before where you'd set pageview as zero?

[the remark this answers is not in the document]
Comment (chrislomaxjones) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts:71]: Oh, unless of course you're not just replicating the logic from before!

**Review by chrislomaxjones — APPROVED** [review]
🏷️

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts]: Actually, @chrislomaxjones, I’ve discovered that we’re not using this value anywhere, so removed in 7d0d98687cca1d2d4e179f852db74d6b7418e6b8. Thanks for taking a closer look!

**Pull request merged**